src/actions/posts.ts

```diff
@@ -417,10 +417,9 @@ export function getUnreadPostData(unreadChan: ChannelUnread, state: GlobalState)
     return data;
 }
 
-export function setUnreadPost(userId: string, postId: string) {
+export function setUnreadPost(userId: string, post: ExtendedPost) {
```

Comment:
Changing the parameters of this function might be a bit tricky since we need to update the web and mobile app code that uses this at the same time that this PR goes in or else it might temporarily break that code. You have a PR open to change the code on the web app that uses this, but we'd need another PR to update the mobile app as well.

Alternatively, instead of changing the parameters so that we can check for `post.system_post_ids`, we might instead be able to call [`isCombinedUserActivityPost`](https://github.com/mattermost/mattermost-redux/blob/master/src/utils/post_list.ts#L192) on the post ID to see what type of post it is.